### PR TITLE
(FACT-2223) hypervisors fact for AIX - alternative implementation

### PR DESCRIPTION
**Pull request by ciprianbadescu**
reimplementation of https://github.com/puppetlabs/facter-ng/pull/209 with separate lpar/wpar facts

### CODE REVIEW

File: lib/facts/aix/hypervisors/lpar.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    class HypervisorsLpar
+      FACT_NAME = 'hypervisors.lpar'
+
+      def call_the_resolver
+        lpar_partition_number = Resolvers::Lpar.resolve(:lpar_partition_number)
+        return ResolvedFact.new(FACT_NAME, nil) unless lpar_partition_number&.positive?
+
+        ResolvedFact.new(FACT_NAME,
+                         partition_number: lpar_partition_number,
+                         partition_name: Resolvers::Lpar.resolve(:lpar_partition_name))
```
Comment (oanatmaria) [inline lib/facts/aix/hypervisors/lpar.rb]: partition_number: lpar_partition_number, partition_name: Resolvers::Lpar.resolve(:lpar_partition_name should be a hash

File: lib/facts/aix/hypervisors/wpar.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+module Facter
+  module Aix
+    class HypervisorsWpar
+      FACT_NAME = 'hypervisors.wpar'
+
+      def call_the_resolver
+        wpar_key = Resolvers::Wpar.resolve(:wpar_key)
+        return ResolvedFact.new(FACT_NAME, nil) unless wpar_key&.positive?
+
+        ResolvedFact.new(FACT_NAME, key: wpar_key, configured_id: Resolvers::Wpar.resolve(:wpar_configured_id))
```
Comment (oanatmaria) [inline lib/facts/aix/hypervisors/wpar.rb]: wrap key and configured_id in a hash

File: lib/facts/aix/hypervisors/wpar.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Facter
+  module Aix
+    class HypervisorsWpar
+      FACT_NAME = 'hypervisors.wpar'
+
+      def call_the_resolver
+        wpar_key = Resolvers::Wpar.resolve(:wpar_key)
+        return ResolvedFact.new(FACT_NAME, nil) unless wpar_key&.positive?
```
Comment (oanatmaria) [inline lib/facts/aix/hypervisors/wpar.rb]: you should verify that oslevel returns an aix variant that is greater than 6, where is the check?

Comment (ciprianbadescu) [reply lib/facts/aix/hypervisors/wpar.rb]: Solved in: https://github.com/puppetlabs/facter-ng/blob/a620243d0141244d43e72ee9871d6fcad5391947/lib/resolvers/wpar_resolver.rb#L21

File: lib/facts/aix/hypervisors/lpar.rb (hunk 1 above)
Comment (ciprianbadescu) [reply lib/facts/aix/hypervisors/lpar.rb]: rubocop issue: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

File: lib/facts/aix/hypervisors/wpar.rb (hunk 2 above)
Comment (ciprianbadescu) [reply lib/facts/aix/hypervisors/wpar.rb]: rubocop issue: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

File: spec/facter/resolvers/lpar_resolver_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+describe 'Lpar' do
+  before do
+    prstat_i = <<-OUTPUT
+  Partition Name                             : aix61-6
+  Partition Number                           : 15
+  Sub Processor Mode                         : -
+    OUTPUT
```
Comment (IrimieBogdan) [inline spec/facter/resolvers/lpar_resolver_spec.rb]: This value can be moved in a file in `spec/fixtures` and accessed with `load_fixture('<file_name>').read`

File: spec/facter/resolvers/wpar_resolver_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe 'Wpar' do
+  context '#oslevel 6.1+' do
+    before do
+      prstat_i = <<-OUTPUT
+    WPAR Key                                   : 13
+    WPAR Configured ID                         : 14
+      OUTPUT
```
Comment (IrimieBogdan) [inline spec/facter/resolvers/wpar_resolver_spec.rb]: This value can be moved in a file in `spec/fixtures` and accessed with `load_fixture('<file_name>').read`

**Review by BogdanIrimie — APPROVED** [review]

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**